src/miner.cpp

```diff
@@ -865,44 +872,37 @@ void BlockAssembler::addPackageTxs(int &nPackagesSelected, int &nDescendantsUpda
                     if (!result.ok) {
                         customTxPassed = false;
                         break;
                     }
 
-                    auto& evmFeeMap = evmPackageContext.feeMap;
-                    auto& evmAddressTxsMap = evmPackageContext.addressTxsMap;
-
-                    const auto addrKey = EvmAddressWithNonce { txResult.sender, txResult.nonce };
-                    if (auto feeEntry = evmFeeMap.find(addrKey); feeEntry != evmFeeMap.end()) {
-                        // Key already exists. We check to see if we need to prioritize higher fee tx
-                        const auto& lastFee = feeEntry->second;
-                        if (txResult.prepay_fee > lastFee) {
+                    const auto evmKey = std::make_pair(txResult.sender, txResult.nonce);
+                    if (evmTXFees.count(evmKey)) {
+                        const auto& gasFees = evmTXFees.at(evmKey);
+                        if (txResult.prepay_fee > gasFees) {
                             // Higher paying fee. Remove all TXs from sender and add to collection to add them again in order.
-                            auto addrTxs = evmAddressTxsMap[addrKey.address];
-                            RemoveTxIters(txIters(addrTxs));
-                            // Remove all fee entries relating to the address
-                            for (auto it = evmFeeMap.begin(); it != evmFeeMap.end();) {
+                            RemoveEVMTransactions(txIters(evmTXs[txResult.sender]));
+                            for (auto it = evmTXFees.begin(); it != evmTXFees.end();) {
                                 const auto& [sender, nonce] = it->first;
-                                if (sender == addrKey.address) {
-                                    it = evmFeeMap.erase(it);
+                                if (sender == txResult.sender) {
+                                    it = evmTXFees.erase(it);
                                 } else {
                                     ++it;
                                 }
                             }
-                            // Buggy code to fix below:
-                            checkedTX.erase(addrTxs[addrKey.nonce]->GetTx().GetHash());
-                            addrTxs[addrKey.nonce] = sortedEntries[i];
-                            auto count{addrKey.nonce};
-                            for (const auto& [nonce, entry] : addrTxs) {
+                            checkedTX.erase(evmTXs[txResult.sender][txResult.nonce]->GetTx().GetHash());
+                            evmTXs[txResult.sender][txResult.nonce] = sortedEntries[i];
+                            auto count{txResult.nonce};
+                            for (const auto& [nonce, entry] : evmTXs[txResult.sender]) {
                                 inBlock.erase(entry);
                                 checkedTX.erase(entry->GetTx().GetHash());
                                 replaceByFee.emplace(count, entry);
```

Comment:
Is there a reason why we do not use the nonce obtained from iterating through (nonce, entry) of the evmTXs map over here?

Reply:
If we're at this point, it reflects that we're in the correct nonce (since validate would not have succeeded otherwise). So, all items in the queue needs to be ordered after.

Reply:
Not the most ideal way, but works for the moment. We'll likely again restructure this in #2220 or later